server.py

```diff
@@ -151,12 +153,20 @@ def get(key):
             os.rename(uri, path.join(settings['assetdir'], asset['asset_id']))
             uri = path.join(settings['assetdir'], asset['asset_id'])
 
+    if 'youtube_asset' in asset['mimetype']:
+        uri, asset['name'] = download_video_from_youtube(uri, asset['asset_id'])
+        asset['mimetype'] = 'video'
+        asset['is_processing'] = 1
+
     asset['uri'] = uri
 
     if "video" in asset['mimetype']:
-        video_duration = get_video_duration(asset['uri'])
-        if video_duration:
-            asset['duration'] = int(video_duration.total_seconds())
+        if asset['is_processing'] == 0:
+            video_duration = get_video_duration(uri)
```

Comment:
I understand it. But now this is how the processing of any video asset works: when adding or editing, the `get_video_duratio`n function works. I just made a video from YouTube a simple video asset.